byway/src/main/webapp/home.js

```diff
@@ -86,22 +87,46 @@ function showIncompleteTrip(tripNum, trip, isDestinationsMissing, isInterestsMis
  * @param {Number} tripNum 
  * @param {Trip} trip
  */
-function showCompleteTrip(tripNum, trip){
+async function showCompleteTrip(trip){
   let container = document.getElementById("past-trips-container");
   let pastTrip = document.createElement('div');
   pastTrip.className = "past-trip";
   let title = document.createElement('a');
+  title.id = "title-" + trip.keyString;
   pastTrip.append(title);
   let mapContainer = document.createElement('div');
   mapContainer.className = 'map';
-  mapContainer.id = "map-" + trip.keyString
+  mapContainer.id = "map-" + trip.keyString;
   pastTrip.append(mapContainer);
   container.append(pastTrip);
   initMap(trip.start, trip.start, trip.route, trip.keyString);
-  title.innerText = "Trip #" + tripNum;
+  let titleString = await constructTripTitle(trip);
+  title.innerText = titleString;
   title.href = configureTripKeyForPath(trip.keyString, "/routepage.html");
 }
 
+async function constructTripTitle(trip){
+  //let title = document.getElementById('title-' + keyString);
+  //title.innerText = "";
+  for(let i = 0; i<30; i++){
+      try{
+        const promises = trip.destinations.map(async (destination) => {
```

Comment:
Oh, I see. You're starting the promises for every destination simultaneously, which is probably why you need to keep increasing the retry budget in `findPlace`. If you do these sequentially instead of simultaneously, you shouldn't need to keep adding more retries inside the `findPlace` function.

Reply:
ah okay i see, I changed it to sequentially and changed the loop to 5 tries